tests/ert/unit_tests/gui/simulation/test_run_dialog.py

```diff
@@ -811,3 +812,50 @@ def qt_bot_click_tab_index(tab_index: int) -> None:
 
     qt_bot_click_tab_index(1)
     assert "Realization id 1 in iteration 1" in fm_step_label.text()
+
+
+@pytest.mark.integration_test
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_design_matrix_alters_num_realizations_field(qtbot: QtBot, storage):
+    xls_filename = "design_matrix.xlsx"
+    design_matrix_df = pd.DataFrame(
+        {
+            "REAL": list(range(3)),
+            "a": [0, 1, 2],
+        }
+    )
+    default_sheet_df = pd.DataFrame([["b", 1], ["c", 2]])
+    with pd.ExcelWriter(xls_filename) as xl_write:
+        design_matrix_df.to_excel(xl_write, index=False, sheet_name="DesignSheet")
+        default_sheet_df.to_excel(
+            xl_write, index=False, sheet_name="DefaultSheet", header=False
+        )
+
+    config_file = "minimal_config.ert"
+    with open(config_file, "w", encoding="utf-8") as f:
+        f.write("NUM_REALIZATIONS 10")
+        f.write(f"\nDESIGN_MATRIX {xls_filename}")
+
+    args_mock = Mock()
+    args_mock.config = config_file
+
+    ert_config = ErtConfig.from_file(config_file)
+    gui = _setup_main_window(ert_config, args_mock, GUILogHandler(), storage)
+    experiment_panel = gui.findChild(ExperimentPanel)
```

Comment:
Did you try parameterizing over this to test all the panels? Might not be trival